ssl/statem/statem_srvr.c

```diff
@@ -1026,9 +985,9 @@ MSG_PROCESS_RETURN tls_process_client_hello(SSL *s, PACKET *pkt)
         /* Load the client random and compression list. */
         challenge_len = challenge_len > SSL3_RANDOM_SIZE ? SSL3_RANDOM_SIZE :
             challenge_len;
-        memset(s->s3->client_random, 0, SSL3_RANDOM_SIZE);
+        memset(clienthello.random, 0, SSL3_RANDOM_SIZE);
```

Comment:
Yes, it should be sizeof(clienthello.random)...and actually the other places around here where we use SSL3_RANDOM_SIZE.